### DISCUSSION TITLE
Shaders: how to import custom shader file in another shader?

### DISCUSSION THREAD

**Discussion by Stranget0** [bevyengine/bevy#12942] [Q&A]
Say I have `assets/shaders/example_shader.wgsl` and want to import stuff from `src/shaders/shader_util.wgsl`. 
In `shader_util` I do `#define_import_path project_name::shader_util;` 
and in `example_shader` I do `#import project_name::shader_util;`, 
but when compiled it fails saying it cannot find the requested module. What am I missing?

**Comment by davids91** [comment]
I think the closest you can get is: https://docs.rs/bevy/latest/bevy/prelude/struct.Shader.html#method.from_wgsl_with_defs

**Comment by jf908** [comment]
The shader material example now shows how to do this: https://bevy.org/examples/shaders/shader-material/

```
#import "shaders/custom_material_import.wgsl"::COLOR_MULTIPLIER
```
